community/kernel/src/main/java/org/neo4j/kernel/impl/store/format/standard/NodeRecordFormat.java

```diff
@@ -68,6 +69,7 @@ public void read( NodeRecord record, PageCursor cursor, RecordLoad mode, int rec
                     BaseRecordFormat.longFromIntAndMod( nextProp, propModifier ), dense,
                     BaseRecordFormat.longFromIntAndMod( nextRel, relModifier ), labels );
         }
+        cursor.setOffset( offset );
```

Comment:
Is it really necessary to save and restore the offset? The `setOffset` method does bounds checking, which would be nice to avoid if possible.

Reply:
Also, I notice that you are not modifying any other standard record formats. I don't think this is necessary because `CommonAbstractStore.prepareForReading` already calls `setOffset` in anticipation of the `shouldRetry` loop.